app/javascript/configureStore.js

```diff
@@ -0,0 +1,29 @@
+import { createStore, applyMiddleware } from "redux";
+import thunk from "redux-thunk";
+
+const initialState = {
+  things: [
+    {
+      name: "test",
+      guild: "123"  
+    }
+  ]
+};
+
+function rootReducer(state, action) {
+  switch (action.type) {
+    case "GET_THINGS_SUCCESS":
+      return { things: action.json.things };
+    default:
+      return state
+  };
+};
+
+export default function configureStore() {
+  const store = createStore(
+    rootReducer,
+    initialState,
+    applyMiddleware(thunk)
+    );
+  return store;
+}
```

Comment:
As requested by the previous code reviewer, instead of having your action in your component, please move them to a separate folder called `redux`. It is an excellent practice always to group your redux in a folder.